doc/how-to/vshard_quick.rst

```diff
@@ -1,160 +1,491 @@
+
+This section describes how to configure the cluster in the ``config.yaml`` file.
+
+..  _vshard-quick-start-configuring-cluster-credentials:
+
+Step 1: Configuring credentials
+*******************************
+
+Add the :ref:`credentials <configuration_reference_credentials>` configuration section:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+    :language: yaml
+    :start-at: credentials:
+    :end-at: roles: [sharding]
+    :dedent:
+
+In this section, two users are created:
+
+*   The ``replicator`` user with the ``replication`` role.
+*   The ``storage`` user with the ``sharding`` role.
+
+
+..  _vshard-quick-start-configuring-cluster-advertise:
+
+Step 2: Specifying advertise URIs
+*********************************
+
+Add the :ref:`iproto.advertise <configuration_reference_iproto_advertise>` section:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+    :language: yaml
+    :start-after: roles: [sharding]
+    :end-at: login: storage
+    :dedent:
+
+In this section, the following options are configured:
+
+*   ``iproto.advertise.peer`` specifies how to advertise the current instance to other cluster members.
+    In particular, this option informs other replica set members that the ``replicator`` user should be used to connect to the current instance.
+*   ``iproto.advertise.sharding`` specifies how to advertise the current instance to a router and rebalancer.
+
+
+..  _vshard-quick-start-configuring-cluster-bucket-count:
+
+Step 3: Configuring bucket count
+********************************
+
+Specify the total number of buckets in a sharded cluster using the ``sharding.bucket_count`` option:
+
+..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+    :language: yaml
+    :start-after: login: storage
+    :end-at: bucket_count
+    :dedent:
+
+
+..  _vshard-quick-start-configuring-cluster-topology:
+
+Step 4: Defining the cluster topology
+*************************************
+
+Define the cluster's topology inside the :ref:`groups <configuration_reference_groups>` section.
+The cluster includes two groups:
+
+*   ``storages`` includes two replica sets. Each replica set contains two instances.
+*   ``routers`` includes one router instance.
+
+Here is a schematic view of the cluster's topology:
+
+.. code-block:: yaml
+
+    groups:
+      storages:
+        replicasets:
+          storage-a:
+            # ...
+          storage-b:
+            # ...
+      routers:
+        replicasets:
+          router-a:
+            # ...
+
+1.  To configure storages, add the following code inside the ``groups`` section:
+
+    ..  literalinclude:: /code_snippets/snippets/sharding/instances.enabled/sharded_cluster/config.yaml
+        :language: yaml
+        :start-at: storages:
+        :end-before: routers:
+        :dedent:
+
+    The main top-level options here are:
```

Comment:
Maybe it is better to say 'group-level' instead of 'top-level' to avoid possible misunderstanding.